aptos-move/block-executor/src/view.rs

```diff
@@ -1267,27 +1266,42 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> LatestView<
                     if !resources_needing_delayed_field_exchange {
                         return None;
                     }
-                    if let Ok(Some(metadata)) = self.get_resource_state_value_metadata(key) {
-                        if let Ok(GroupReadResult::Size(group_size)) =
-                            unsync_map.get_group_size(key)
-                        {
-                            return Some(Ok((key.clone(), (metadata, group_size.get()))));
-                        } else {
-                            // TODO[agg_v2](cleanup): `get_group_size` can fail on group tag serialization. Do
-                            //       we want to propagate this error? This is somewhat an invariant
-                            //       violation so PanicError is also ok?
-                            return Some(Err(code_invariant_error(format!(
-                                "Cannot compute metadata op size for the group read {:?}",
-                                key
-                            ))));
+                    match self.get_resource_state_value_metadata(key) {
```

Comment:
Should we just `?` here to avoid `Err(e)` case below and avoid remapping? This will also make this more similar to `get_group_reads_needing_exchange_parallel` implementation

Reply:
sequential should not fail ever, but let me change it for consistency